ejb3/src/main/java/org/jboss/as/ejb3/deployment/processors/BusinessViewAnnotationProcessor.java

```diff
@@ -197,12 +211,18 @@ private void processViewAnnotations(final DeploymentUnit deploymentUnit, final C
         return Arrays.asList(localViews);
     }
 
-    private static Collection<Class<?>> getBusinessInterfacesFromInterfaceAnnotations(Class<?> sessionBeanClass, Class<? extends Annotation> annotation) {
+    private static Collection<Class<?>> getBusinessInterfacesFromInterfaceAnnotations(Class<?> sessionBeanClass, Class<? extends Annotation> annotation) throws DeploymentUnitProcessingException {
         final Set<Class<?>> potentialBusinessInterfaces = getPotentialBusinessInterfaces(sessionBeanClass);
         final Set<Class<?>> businessInterfaces = new HashSet<Class<?>>();
         for (Class<?> iface : potentialBusinessInterfaces) {
-            if (iface.getAnnotation(annotation) != null) {
-                businessInterfaces.add(iface);
+            try {
+                if (iface.getAnnotation(annotation) != null) {
+                    businessInterfaces.add(iface);
+                }
+            } catch (ArrayStoreException e) {
+                // https://bugs.openjdk.java.net/browse/JDK-7183985
+                // Class.findAnnotation() has a bug under JDK < 11 which throws ArrayStoreException
+                throw EjbLogger.ROOT_LOGGER.missingClassInAnnotation(annotation.getSimpleName(), iface.getName());
```

Comment:
I suspect we don't need exception handling here. When `@Remote @Local` is used in a business interface, they do not take any arguments.  The no-arg annotation is to indicate that the annotated business interface is a local or remote business interface.  For ex,

```java
@Remote 
public interface Calculator {...}
```

Reply:
I think it should be good if users do it in correct way(like they should package classes referred by the annotation). :)
but what happens if users referred some classes in the business interfaces' Remote annotation by mistake? is it still a valid deployment?

Reply:
If `@Remote` or `@Local` on business interface has param, then it's an application's error, though the spec does not say what should happen in this case. WildFly just ignores the param, which happens after the annotation processing. So to tolerate this type of user errors, it's okay to keep this checking.